pkg/kv/kvserver/replica.go

```diff
@@ -436,6 +436,46 @@ type Replica struct {
 	// timestamp, independent of the source.
 	sideTransportClosedTimestamp sidetransportAccess
 
+	// shMu contains "shared" fields which are mutated while both raftMu and mu are
+	// held. They can be accessed when either of the two mutexes is held.
+	//
+	// TODO(pav-kv): audit all other fields and include here.
+	shMu struct {
```

Comment:
```
"Do not communicate by sharing memory; instead, share memory by communicating."
```

Reply:
Much better! Thank you for slogging through another mass rename. I like where you got this.

Reply:
Not going to review this commit line-by-line on the assumption that it's 100% mechanical - let me know if there are parts that could use a human check.

Reply:
The big commits are 100% mechanical. The "interesting" ones (where I removed unnecessary locking) are separate and small, so hopefully you reviewed them.